File: fluentcore/src/main/java/com/mcreater/fxfluent/stage/controls/FluentWindowButton.java
package com.mcreater.fxfluent.stage.controls;

import com.mcreater.fxfluent.controls.FluentButton;
import com.mcreater.fxfluent.xaml.style.SystemThemeLoop;
import javafx.scene.Node;
import javafx.scene.control.Button;

public class FluentWindowButton extends Button {
    public FluentWindowButton() {
        this.init();
    }

    private FluentWindowButton(String var1) {
        super(var1);
        this.init();
    }

    public FluentWindowButton(String var1, Node var2) {
        super(var1, var2);
        this.init();
    }
    private void init() {
        getStyleClass().add("fluent-window-button");
        setPrefSize(48, 34);
        setMaxSize(48, 34);
        setMinSize(48, 34);
    }
    public String getUserAgentStylesheet() {
        return FluentButton.class.getClassLoader().getResource(SystemThemeLoop.isDark() ? "css/FluentWindowButtonDark.css" : "css/FluentWindowButtonLight.css").toString();
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/stage/controls/FluentWindowButton.kt
package com.mcreater.fxfluent.stage.controls

import com.mcreater.fxfluent.controls.FluentButton
import com.mcreater.fxfluent.xaml.style.SystemThemeLoop
import javafx.scene.control.Button

class FluentWindowButton: Button() {
    init {
        init()
    }
    private fun init() {
        styleClass.add("fluent-window-button")
        setPrefSize(48.0, 34.0)
        setMaxSize(48.0, 34.0)
        setMinSize(48.0, 34.0)
    }

    override fun getUserAgentStylesheet(): String {
        return FluentButton::class.java.getClassLoader()
            .getResource(if (SystemThemeLoop.isDark()) "css/FluentWindowButtonDark.css" else "css/FluentWindowButtonLight.css")
            ?.toString() ?: ""
    }
}